src/lib_io.c

```diff
@@ -164,9 +164,14 @@ static void io_file_readall(lua_State *L, FILE *fp)
   MSize m, n;
   for (m = LUAL_BUFFERSIZE, n = 0; ; m += m) {
     char *buf = lj_buf_tmp(L, m);
-    n += (MSize)fread(buf+n, 1, m-n, fp);
-    if (n != m) {
-      setstrV(L, L->top++, lj_str_new(L, buf, (size_t)n));
+    for (;;) {
+      n += (MSize)fread(buf+n, 1, m-n, fp);
+      if (n == m) break;
+      else if (ferror(fp)) {
```

Comment:
We can save the `else` word here since the previous branch never falls through.

Reply:
This branch is to prevent adding the read data to the stack upon an error, because I believe the previous behavior was an oversight due to the previous design of this function, which had no error handling at all.

Reply:
@thibaultcha OK, I see we now have proper error handling for general file I/O errors. Huh, more than I expected :)

Reply:
@thibaultcha The `else` here does not make any difference, no? the previous branch never falls through (it always breaks).

Reply:
The else branch prevents adding the read data to the stack in case of error (if we want this behavior, which I think we do). It is to run if and only if `ferror(fp) == 0` (and, inheritently, when `n != m`).
If we remove it, and an error occurs that is not EINTR, we would add the read data to the stack, run the GC, and return. I am trying to not add the read data to the stack (because it is never used in the caller’s code path upon error), and still perform the last 2 steps (GC + return).

Reply:
Sorry, on mobile (still) I just realized this is the above branch you are talking about. I’ll gladly remove the `else` here :)

Reply:
Removed!